File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        if not word1: return word2
        if not word2: return word1
        
        p1 = p2 = 0
        res = ''
        while p1 < len(word1) and p2 < len(word2):
            res += word1[p1]
            p1 += 1
            res += word2[p2]
            p2 += 1
        
        if p1 < len(word1):
            while p1 < len(word1):
                res += word1[p1]
                p1 += 1
        
        if p2 < len(word2):
            while p2 < len(word2):
                res += word2[p2]
                p2 += 1
        
        return res

File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res = "";
        int w1 = 0, w2 = 0;
        while(w1 < word1.length() || w2 < word2.length()) {
            if(w1 == word1.length()) {
                while(w2 < word2.length()) {
                    res = res + word2[w2];
                    w2++;
                }
            }
            else if(w2 == word2.length()) {
                while(w1 < word1.length()) {
                    res = res + word1[w1];
                    w1++;
                }
            }
            else {
                res = res + word1[w1] + word2[w2];
                w1++;
                w2++;
            }
            
        }
        
        return res;
    }
};